package.json

```diff
@@ -77,6 +74,11 @@
     "husky": "^4.0.10",
     "lint-staged": "^10.0.2"
   },
+  "peerDependencies": {
+    "go-ipfs-dep": "*",
+    "ipfs": "*",
+    "ipfs-http-client": "*"
```

Comment:
...but I think you'll have to install all of these even if you use only 1 because the calls to require will fail? Do we need try/catches?

Reply:
Quite right.  I think you'll need `ipfs-http-client` all the time, even the `proc` factory type needs it because it checks to see if another ipfs process has opened the repo, if it has then it uses the http client to connect to it.

I've pushed a change that only requires `ipfs` if we're going to use it.  `go-ipfs-dep` isn't required directly, instead the `findBin` function in `utils.js` tries to pull the binary path out of the module so I've made that fail silently if the factory type is not `'go'`.